src/bridges/aave/lending/AaveLendingBridge.sol

```diff
@@ -0,0 +1,342 @@
+     * @param totalInputValue The input amount of inputAssetA
+     * @param interactionNonce The interaction nonce of the call
+     * @param auxData Unused auxiliary information
+     * @return outputValueA The output amount of outputAssetA
+     * @return outputValueB The ouput amount of outputAssetB
+     * @return isAsync Always false for this bridge
+     */
+    function convert(
+        AztecTypes.AztecAsset calldata inputAssetA,
+        AztecTypes.AztecAsset calldata inputAssetB,
+        AztecTypes.AztecAsset calldata outputAssetA,
+        AztecTypes.AztecAsset calldata outputAssetB,
+        uint256 totalInputValue,
+        uint256 interactionNonce,
+        uint64 auxData,
+        address rollupBeneficiary
+    )
+        external
+        payable
+        override(IDefiBridge)
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool isAsync
+        )
+    {
+        (bool enter, address underlyingAddress, address zkATokenAddress, bool isEth) = _sanityConvert(
+            inputAssetA,
+            inputAssetB,
+            outputAssetA,
+            outputAssetB
+        );
+
+        if (enter) {
+            outputValueA = _enter(underlyingAddress, zkATokenAddress, totalInputValue, isEth);
+        } else {
+            outputValueA = _exit(underlyingAddress, zkATokenAddress, totalInputValue, interactionNonce, isEth);
+        }
+
+        return (outputValueA, 0, false);
+    }
+
+    /**
+     * @notice Deposit into Aave with `amount` of `underlyingAsset` and return the corresponding amount of zkATokens
+     * @param underlyingAsset The address of the underlying asset
+     * @param amount The amount of underlying asset to deposit
+     * @return The amount of zkAToken that was minted by the deposit
+     */
+    function _enter(
+        address underlyingAsset,
+        address zkATokenAddress,
+        uint256 amount,
+        bool isEth
+    ) internal returns (uint256) {
+        /**
+         * Interaction flow:
+         * 0. If receiving ETH, wrap it such that WETH can be deposited
+         * 1. Fetch current liquidity index from Aave and compute scaled amount
+         * 2. Approve underlying asset to be deposited into Aave
+         * 3. Deposit assets into Aave (receives aUnderlyingAsset in return)
+         * 4. Mint zkATokens equal to scaled amount
+         * 5. Approve ROLLUP_PROCESSOR to pull funds
+         */
+
+        if (isEth) {
+            WETH.deposit{value: amount}();
+        }
+
+        ILendingPool pool = ILendingPool(ADDRESSES_PROVIDER.getLendingPool());
+        // Compute scaledAmount rounded down. Aave uses rayDiv which can round up and down (at 0.5).
+        // For consistency we round down.
+        uint256 scaledAmount = (amount * 1e27) / pool.getReserveNormalizedIncome(underlyingAsset);
+        require(scaledAmount > 0, Errors.ZERO_VALUE);
+
+        IERC20(underlyingAsset).safeApprove(address(pool), amount);
```

Comment:
We trust the pool with the underlying so pre-approving makes sense here as well.

Reply:
Approvals moved to a separate function to ensure that we can re-approve in case Aave updates their proxy address for some reason. Approvals performed when listing asset initially as well. See https://github.com/AztecProtocol/aztec-connect-bridges/pull/21/commits/7053a154f0360e3eadd3ed3a1f9403fff90854f9